docs/literate/src/files/subcell_shock_capturing.jl

```diff
@@ -0,0 +1,285 @@
+
+# In the previous tutorial, the element-wise limiting with [`IndicatorHennemannGassner`](@ref)
+# and [`VolumeIntegralShockCapturingHG`](@ref) was explained. This tutorial contains a short
+# introduction to the idea and implementation of subcell shock capturing approaches in Trixi.jl,
+# which is also based on the DGSEM scheme in flux differencing formulation.
+# Trixi.jl contains the a-posteriori invariant domain-preserving (IDP) limiter which was
+# introduced by [Pazner (2020)](https://doi.org/10.1016/j.cma.2021.113876) and
+# [Rueda-Ramírez, Pazner, Gassner (2022)](https://doi.org/10.1016/j.compfluid.2022.105627).
+# It is a flux-corrected transport-type (FCT) limiter and is implemented using [`SubcellLimiterIDP`](@ref)
+# and [`VolumeIntegralSubcellLimiting`](@ref).
+# Since it is an a-posteriori limiter you have to apply a correction stage after each Runge-Kutta
+# stage. This is done by passing the stage callback [`SubcellLimiterIDPCorrection`](@ref) to the
+# time integration method.
+
+# ## Time integration method
+# As mentioned before, the IDP limiting is an a-posteriori limiter. Its limiting process
+# guarantees the target bounds for an explicit (forward) Euler time step. To still achieve a
+# high-order approximation, the implementation uses strong-stability preserving (SSP) Runge-Kutta
+# methods, which can be written as convex combinations of forward Euler steps.
+#-
+# Since IDP/FCT limiting procedure operates on independent forward Euler steps, its
+# a-posteriori correction stage is implemented as a stage callback that is triggered after each
+# forward Euler step in an SSP Runge-Kutta method. Unfortunately, the `solve(...)` routines in
+# [OrdinaryDiffEq.jl](https://github.com/SciML/OrdinaryDiffEq.jl), typically employed for time
+# integration in Trixi.jl, do not support this type of stage callback.
+#-
+# Therefore, subcell limiting with the IDP limiter requires the use of a Trixi-intern
+# time integration SSPRK method called with
+# ````julia
+# Trixi.solve(ode, method(stage_callbacks = stage_callbacks); ...)
+# ````
+#-
+# Right now, only the canonical three-stage, third-order SSPRK method (Shu-Osher)
+# [`Trixi.SimpleSSPRK33`](@ref) is implemented.
+
+# # [IDP Limiting](@id IDPLimiter)
+# The implementation of the invariant domain preserving (IDP) limiting approach ([`SubcellLimiterIDP`](@ref))
+# is based on [Pazner (2020)](https://doi.org/10.1016/j.cma.2021.113876) and
+# [Rueda-Ramírez, Pazner, Gassner (2022)](https://doi.org/10.101/j.compfluid.2022.105627).
+# It supports several types of limiting which are enabled by passing parameters individually.
+
+# ### [Global bounds](@id global_bounds)
+# If enabled, the global bounds enforce physical admissibility conditions, such as non-negativity
+# of variables. This can be done for conservative variables, where the limiter is of a one-sided
+# Zalesak-type, and general non-linear variables, where a Newton-bisection algorithm is used to
```

Comment:
Maybe reference for "Zalesak-type" (never heard of this)

Reply:
Yes a good idea. I added it in [cf49c37](https://github.com/trixi-framework/Trixi.jl/pull/1882/commits/cf49c377c88cb64175335e32815c8f1fab0b4389)